### ISSUE TITLE
use auc as monitor property in ModelCheckpoint

### ISSUE THREAD

**Issue by yesufeng** [fchollet/keras#5170]
is there a way to use validation data auc as monitor in ModelCheckpoint? I want to save the model that has the best validation set auc. Thanks!

- [ x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by kgrm** [comment]
You will need to define an auc metric, similarly to those in https://github.com/fchollet/keras/blob/master/keras/metrics.py . Then, add it to the list of metrics when compiling your model, and pass it to the ModelCheckpoint initialisation, with mode="max".

**Comment by yesufeng** [comment]
Thanks!

**Comment by jdelange** [comment]
@yesufeng Did you create an auc metric that you would be willing to share?

**Comment by jdelange** [comment]
Doing some more searching, I don't think using metrics is the way to go: [#3230](https://github.com/fchollet/keras/issues/3230)

**Comment by kgrm** [comment]
It's possible, but you'd have to write the AUC routine yourself using the keras backend, you can't rely on external libraries.

**Comment by jdelange** [comment]
Following #3230, I ended up following this example [Callback example](https://github.com/keunwoochoi/keras_callbacks_example), whereby replaced the lists[] with a max_score per epoch. If exceeded, then I save the weights of that epoch and update max_score. This potentially involves a disc access every epoch. Ideally you would store the best weights as a variable and save once at on_train_end, but I haven't figured out how to do that yet.

**Comment by yesufeng** [comment]
That's what I used as well, define a customized callback function to save the best auc.
